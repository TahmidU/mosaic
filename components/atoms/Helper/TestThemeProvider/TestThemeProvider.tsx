import { ReactElement } from "react";
import { Theme } from "types/context";
import { ThemeProvider } from "styled-components";
import getTheme from "resources/themes";

interface TestThemeProviderProps {
  children: any;
  theme?: Theme;
}

export default function TestThemeProvider({
  children,
  theme = "light",
}: TestThemeProviderProps): ReactElement {
  return <ThemeProvider theme={getTheme(theme)}>{children}</ThemeProvider>;
}
