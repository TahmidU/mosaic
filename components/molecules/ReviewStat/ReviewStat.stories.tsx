import { IPayload } from "types/story";
import ReviewStat from ".";

const payload: IPayload = {
  title: "MOLECULES/ReviewStat",
  component: ReviewStat,
};
export default payload;

export const Basic = (args: any) => <ReviewStat {...args} />;
Basic.args = { percentage: 60 };
