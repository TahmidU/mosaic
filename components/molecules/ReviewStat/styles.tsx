import { CircularProgressbar } from "react-circular-progressbar";
import styled from "styled-components";

export const Container = styled.div`
  background-color: ${({ theme }) => theme.almostBlack};
  border-radius: 100%;
  width: 160px;
  height: 160px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid ${({ theme }) => theme.almostBlack};
`;
Container.displayName = "Container";

export const ProgCircle = styled(CircularProgressbar)`
  width: 100%;
  height: 100%;

  path {
    stroke-width: 5;
  }

  > text:last-child {
    fill: ${({ theme }) => theme.white};
    ${({ theme }) => theme.fonts.main.bigger};
    font-weight: bolder;
    dominant-baseline: middle;
    text-anchor: middle;
  }
`;
ProgCircle.displayName = "ProgCircle";
