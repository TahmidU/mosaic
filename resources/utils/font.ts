/**
 *
 * @param font : font from IFonts;
 * @param indexes : 0 - font-family and 1 - font-size
 * @returns font-family + font-size string
 */
export function fontImportancePlacer(font: string, indexes: number[]): string {
  return font
    .split(";")
    .map((f, index) => (indexes.includes(index) ? f + " !important" : f))
    .join(";");
}
