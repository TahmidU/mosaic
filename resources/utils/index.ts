export { default as TextUtils } from "./TextUtils";
export { default as StoryUtils } from "./StoryUtils";
