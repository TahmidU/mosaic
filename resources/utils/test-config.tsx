import React, { FC, ReactElement } from "react";
import { RenderOptions, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import getTheme from "../themes";

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={getTheme("light")}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
