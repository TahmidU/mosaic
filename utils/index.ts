export { default as TextUtils } from "./TextUtils";
