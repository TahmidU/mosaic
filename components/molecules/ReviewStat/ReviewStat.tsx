import { Container, ProgCircle, TextStyle } from "./styles";
import { ReactElement, useState } from "react";

interface ReviewStatProps {}

export default function ReviewStat({}: ReviewStatProps): ReactElement {
  const [percentage, setPercentage] = useState(0);

  return (
    <Container>
      <ProgCircle width={160} height={160} radius={80} strokeWidth={10} />
      <TextStyle> {`${percentage}%`} </TextStyle>
    </Container>
  );
}
