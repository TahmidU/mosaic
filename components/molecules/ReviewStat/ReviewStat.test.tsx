import "jest-styled-components";

import { cleanup, render } from "resources/utils/test-config";

import React from "react";
import ReviewStat from "./ReviewStat";
import TestThemeProvider from "components/atoms/TestThemeProvider";
import getTheme from "resources/themes";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("ReviewStat", () => {
  test("Review 0%", () => {
    // Given When
    const review = 0;
    const { getByText } = render(<ReviewStat percentage={review} />);

    // Then
    getByText(`${review}%`);
  });

  test("Review 90% non-wait", () => {
    // Given When
    const review = 90;
    const { queryByText } = render(<ReviewStat percentage={review} />);

    // Then
    expect(queryByText(`${review}%`)).toBe(null);
  });

  test("Review 90%", async () => {
    // Given When
    const review = 90;
    const { getByText } = render(<ReviewStat percentage={review} />);

    // Then
    await new Promise((r) => setTimeout(r, 1400));
    getByText(`${review}%`);
  });

  test("Default light theme styles applied", async () => {
    // Given
    const lightTheme = getTheme("light");
    const review = 10;

    // When
    const { getByTestId } = render(<ReviewStat percentage={review} />);

    // Then
    expect(getByTestId("ReviewStatContainer")).toHaveStyleRule(
      "background-color",
      lightTheme.almostBlack
    );
    expect(getByTestId("ReviewStatContainer")).toHaveStyleRule(
      "border-radius",
      "100%"
    );
    expect(getByTestId("ReviewStatContainer")).toHaveStyleRule(
      "border",
      `1px solid ${lightTheme.almostBlack}`
    );
  });

  test("Snapshot", () => {
    // Given
    const review = 10;

    // When
    const rendered = renderer.create(
      <TestThemeProvider>
        <ReviewStat percentage={review} />
      </TestThemeProvider>
    );
    const tree = rendered.toJSON();

    // Then
    expect(tree).toMatchSnapshot();
  });
});
