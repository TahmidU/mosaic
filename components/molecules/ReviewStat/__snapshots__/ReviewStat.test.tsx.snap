// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewStat Snapshot 1`] = `
.c0 {
  background-color: #211f1b;
  border-radius: 100%;
  width: 160px;
  height: 160px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid #211f1b;
}

.c1 {
  width: 100%;
  height: 100%;
}

.c1 path {
  stroke-width: 5;
}

.c1 > text:last-child {
  fill: #FFFFFF;
  font-family: "Roboto",sans-serif;
  font-size: 24px;
  font-weight: bolder;
  -webkit-dominant-baseline: middle;
  dominant-baseline: middle;
  text-anchor: middle;
}

<div
  className="c0"
  data-testid="ReviewStatContainer"
>
  <svg
    className="CircularProgressbar c1"
    data-test-id="CircularProgressbar"
    style={Object {}}
    viewBox="0 0 100 100"
  >
    <path
      className="CircularProgressbar-trail"
      d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
      fillOpacity={0}
      strokeWidth={8}
      style={
        Object {
          "stroke": "#e8e8e8",
          "strokeDasharray": "289.02652413026095px 289.02652413026095px",
          "strokeDashoffset": "0px",
        }
      }
    />
    <path
      className="CircularProgressbar-path"
      d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
      fillOpacity={0}
      strokeWidth={8}
      style={
        Object {
          "stroke": "rgba(56, 255, 76, 1)",
          "strokeDasharray": "289.02652413026095px 289.02652413026095px",
          "strokeDashoffset": "289.02652413026095px",
          "transition": "none",
        }
      }
    />
    <text
      className="CircularProgressbar-text"
      style={Object {}}
      x={50}
      y={50}
    >
      0%
    </text>
  </svg>
</div>
`;
